commit 19a198f05802fcc05441c364ed75311ef3f6d663
Author: Siddhesh Poyarekar <siddhesh@sourceware.org>
Date:   Thu Nov 28 06:30:40 2024 -0500

    pthread_getcpuclockid: Add descriptive comment to smoke test
    
    Add a descriptive comment to the tst-pthread-cpuclockid-invalid test and
    also drop pthread_getcpuclockid from the TODO-testing list since it now
    has full coverage.
    
    Signed-off-by: Siddhesh Poyarekar <siddhesh@sourceware.org>

diff --git a/nptl/TODO-testing b/nptl/TODO-testing
index e076e5624f1cfbaa..f50d2ceb51b247c3 100644
--- a/nptl/TODO-testing
+++ b/nptl/TODO-testing
@@ -10,10 +10,6 @@ pthread_attr_[sg]etstack
 
   some more tests needed
 
-pthread_getcpuclockid
-
-  check that value is reset -> rt subdir
-
 pthread_getschedparam
 pthread_setschedparam
 
diff --git a/nptl/tst-pthread-getcpuclockid-invalid.c b/nptl/tst-pthread-getcpuclockid-invalid.c
index e88a56342767a83e..7ac46acad8fe0fd7 100644
--- a/nptl/tst-pthread-getcpuclockid-invalid.c
+++ b/nptl/tst-pthread-getcpuclockid-invalid.c
@@ -1,5 +1,4 @@
-/* Smoke test to verify that pthread_getcpuclockid fails with ESRCH when the
-   thread in question has exited.
+/* pthread_getcpuclockid should fail with ESRCH when the thread exits.
    Copyright the GNU Toolchain Authors.
    This file is part of the GNU C Library.
 
@@ -17,6 +16,12 @@
    License along with the GNU C Library; if not, see
    <https://www.gnu.org/licenses/>.  */
 
+/* The input thread descriptor to pthread_getcpuclockid needs to be valid when
+   the function is called.  For the purposes of this test, this means that the
+   thread should not be detached, have exited, but not joined.  This should be
+   good enough to complete coverage for pthread_getcpuclockid alongside
+   tst-clock2.  */
+
 #include <errno.h>
 #include <pthread.h>
 #include <sched.h>
